diesel/src/pg/metadata_lookup.rs

```diff
@@ -0,0 +1,32 @@
+use std::mem;
+
+use prelude::*;
+use super::{PgConnection, PgTypeMetadata};
+
+#[allow(missing_debug_implementations)]
+pub struct PgMetadataLookup {
+    conn: PgConnection,
+}
+
+impl PgMetadataLookup {
+    pub(crate) fn new(conn: &PgConnection) -> &Self {
+        unsafe { mem::transmute(conn) }
```

Comment:
I generally prefer to avoid code comments which can get out of sync with the code they're referencing. I explained the reasoning for this in the commit message, which is where I prefer to document these things. This function doesn't have any invariants which we could violate without rewriting the function.